src/main/java/me/contaria/seedqueue/gui/wall/SeedQueueWallScreen.java

```diff
@@ -85,9 +85,10 @@ public class SeedQueueWallScreen extends Screen {
 
     public SeedQueueWallScreen(Screen createWorldScreen) {
         super(LiteralText.EMPTY);
+        this.layout = Layout.createLayout();
```

Comment:
Creating the layout in the constructor isn't necessary, there is no need to set an initial capacity for the `preparingPreviews` list, just leave it default

Reply:
> Creating the layout in the constructor isn't necessary, there is no need to set an initial capacity for the `preparingPreviews` list, just leave it default

having it empty causes a bug which stops preparing previews from moving into main grid (only the intial main grid seeds are there and after that no more seeds come up there)

Reply:
That is definitely a different bug, the arraylists initial capacity does not matter for any logic. Per it's Javadoc, "As elements are added to an ArrayList, its capacity grows automatically."

Reply:
> That is definitely a different bug, the arraylists initial capacity does not matter for any logic. Per it's Javadoc, "As elements are added to an ArrayList, its capacity grows automatically."

ok i've tested the second time and seems like you're right